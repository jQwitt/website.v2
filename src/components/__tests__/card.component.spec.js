import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Card from '../card.component';

afterEach(cleanup);
describe('Card renders correctly', () => {
  test('snapshot', () => {
    const { asFragment } = render(<Card />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('conditionally renders title', () => {
    const data = {
      title: 'Title',
    };
    const { getByText } = render(<Card title={data.title} />);
    expect(getByText(data.title)).toBeTruthy();
  });
  it('conditionally renders subTitle', () => {
    const data = {
      subTitle: 'description',
    };
    const { getByText } = render(<Card title={data.subTitle} />);
    expect(getByText(data.subTitle)).toBeTruthy();
  });
  test('renders all props correctly', () => {
    const data = {
      title: 'Title',
      subTitle: 'description',
    };
    const { getByText } = render(
      <Card title={data.title} subTitle={data.subTitle} />,
    );
    expect(getByText(data.title)).toBeTruthy();
    expect(getByText(data.subTitle)).toBeTruthy();
  });
});
