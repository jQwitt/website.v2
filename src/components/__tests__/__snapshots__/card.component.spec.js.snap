// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders correctly snapshot 1`] = `
<DocumentFragment>
  <div
    class="col-12 col-md-6 col-lg-4 col-xl-4 content-card"
  >
    <div
      class="img"
    />
    <div
      class="text"
    />
  </div>
</DocumentFragment>
`;
